doc/devdocs/development/new-powertoy.md

```diff
@@ -0,0 +1,294 @@
+# 🧭 Creating a New PowerToy: End-to-End Developer Guide
+
+First of all, thank you for wanting to contribute to PowerToys. The work we do would not be possible with out the support of community supporters like you.
+
+This guide documents the process of building a new PowerToys utility from scratch, including architecture decisions, integration steps, and common pitfalls.
+
+---
+
+## 1. Overview and Prerequisites
+
+A PowerToy module is a self-contained utility integrated into the PowerToys ecosystem. It can be UI-based, service-based, or both.
+
+### Requirements
+
+* [Visual Studio 2022](https://visualstudio.microsoft.com/downloads/) and the following workloads/individual components
+  - Desktop Development with C++
+  - WinUI application development
+  - .NET desktop development
+  - Windows 10 SDK (10.0.22621.0)
+  - Windows 11 SDK (10.0.26100.3916)
+* .NET 8 SDK
+* Clone the [PowerToys repository](https://github.com/microsoft/PowerToys/tree/main) locally
+* [Validate that you are able to build and run](https://github.com/microsoft/PowerToys/blob/main/doc/devdocs/development/debugging.md) `PowerToys.sln`.
+
+Optional: 
+* [WiX v5 toolset](https://github.com/microsoft/PowerToys/tree/main) for the installer
+
+> Note: To ensure all the correct VS Workloads are installed, use [the WinGet configuration files](https://github.com/microsoft/PowerToys/tree/e13d6a78aafbcf32a4bb5f8581d041e1d057c3f1/.config) in the project repository. (Use the one that matches your VS distribution. ie: VS Enterprise would use `configuration.vsEnterprise.winget`)
+
+### Folder Structure
+
+```
+src/
+  modules/
+    your_module/
+      YourModule.sln
+      YourModuleInterface/
+      YourModuleUI/ (if needed)
+      YourModuleService/ (if needed)
+```
+
+---
+## 2. Design and Planning
+
+### Decide the Type of Module
+Think about how your module works and which existing modules behave similarly. You are going to want to think about the UI needed for the application, the lifecycle, whether it is a service that is always running or event based. Below are some basic scenarios with some modules to explore. You can write your application in C++ or C#.
+* **UI-only:** e.g., ColorPicker
+* **Background service:** e.g., LightSwitch, Awake
+* **Hybrid (UI + background logic):** e.g., ShortcutGuide
+* **C++/C# interop:** e.g., PowerRename 
+
+### Writing your Module Interface
+Begin by setting up the [PowerToy module template project](https://github.com/microsoft/PowerToys/tree/main/tools/project_template). This will generate boilerplate for you to begin your new module. Below I will go through the key headers in the Module Interface (dllmain.cpp) and explain their purpose:
```

Comment:
```suggestion
Begin by setting up the [PowerToy module template project](https://github.com/microsoft/PowerToys/tree/main/tools/project_template). This will generate boilerplate for you to begin your new module. Below are the key headers in the Module Interface (`dllmain.cpp`) and an explanation of their purpose:
```